tortoise/tortoise.go

```diff
@@ -387,16 +369,16 @@ func (t *turtle) encodeVotes(
 // getFullVote unlike getLocalVote will vote according to the counted votes on blocks that are
 // outside of hdist. if opinion is undecided according to the votes it will use coinflip recorded
 // in the current layer.
-func (t *turtle) getFullVote(ctx context.Context, block *blockInfo) (sign, voteReason, error) {
-	vote, reason := getLocalVote(&t.state, t.Config, block)
+func (t *turtle) getFullVote(verified, current types.LayerID, block *blockInfo) (sign, voteReason, error) {
+	vote, reason := getLocalVote(verified, current, t.Config, block)
 	if !(vote == abstain && reason == reasonValidity) {
 		return vote, reason, nil
 	}
 	vote = sign(block.margin.Cmp(t.localThreshold))
 	if vote != abstain {
 		return vote, reasonLocalThreshold, nil
 	}
-	coin, err := layers.GetWeakCoin(t.cdb, t.last)
+	coin, err := layers.GetWeakCoin(t.cdb, current.Sub(1))
```

Comment:
i don't understand how this closes https://github.com/spacemeshos/go-spacemesh/issues/2920

shouldn't this be `coin, err := layers.GetWeakCoin(t.cdb, block.layer.Sub(1))`?

Reply:
current is a layer when ballot is created, so it uses weak coin from that was collected right before

i am not aware how block is related to weak coin, is it? or that was a ballot/block typo?

Reply:
it is a type. plz do `:s/block/ballot/` in my msg.

the confusing part for me is that in the call chain of `getFullVote` the variable is called `last`, which was updated with the current layer in the miner pkg. will it make sense to rename them all to current?

Reply:
updated